services/minespace-web/src/components/pages/Incidents/IncidentPage.js

```diff
@@ -144,16 +170,82 @@ const StepForms = (props, state, navigation, handlers, formatInitialValues) => [
         >
           Back
         </Button>
-        <Button id="step-2-next" type="primary" disabled onClick={() => {}}>
+        <Button
+          id="step-2-next"
+          type="primary"
+          disabled={disabledButton}
+          onClick={async (e) => {
+            const response = await handlers?.save(e, props.formValues);
+            const incidentGuid = props.incident?.mine_incident_guid || response?.mine_incident_guid;
+            const mineGuid = props.incident?.mine_guid || response?.mine_guid;
+            if (incidentGuid && mineGuid) {
+              return props.history.push({
+                pathname: `${routes.REVIEW_MINE_INCIDENT.dynamicRoute(mineGuid, incidentGuid)}`,
+                state: { current: 2 },
+              });
+            }
+            return null;
+          }}
+        >
           Review & Submit
         </Button>
       </React.Fragment>,
     ],
   },
   {
     title: "Review & Submit",
-    content: <></>,
-    buttons: [],
+    content: (
+      <IncidentReviewSubmit
+        setConfirmedSubmission={setConfirmedSubmission}
+        confirmedSubmission={state.confirmedSubmission}
+        incident={props.incident}
+        formValues={props.formValues}
+      />
+    ),
+    buttons: [
+      <React.Fragment key="step-3-buttons">
+        <Button
+          id="step-back2"
```

Comment:
Not a big issue, but this id seems to reference step 2. Wouldn't worry about it unless you are updating other stuff in this component.